### DISCUSSION TITLE
Cannot find a remote target in Chrome while debugging the app with Hermes

### DISCUSSION THREAD

**Discussion by JiglioNero** [microsoft/react-native-windows#9087] [General]
Hi I am a member of the [`vscode-react-native`](https://github.com/microsoft/vscode-react-native) development team. We tested Hermes debugging of the recent versions of React Native for Windows and found out that for `react-native-windows v0.66.2` and newer, we couldn't find a debuggable target for the app in the Google Chrome inspect page (`chrome://inspect`). We configure the project in accordance with official guides: [here](https://microsoft.github.io/react-native-windows/docs/getting-started) and [here](https://microsoft.github.io/react-native-windows/docs/hermes). Here is our steps to reproduce this behavior:
1. Initialize a new RN application using the `npx react-native init <projectName> --template react-native@^0.66.0` command
2. Initialize react-native-windows using the `npx react-native-windows-init --overwrite --useHermes` command
3. After the process is complete, run the RNW application in direct debugging mode using the `npx react-native run-windows --direct-debugging` command
4. Open Google Chrome inspect page (`chrome://inspect`) and check for debuggable targets

![image](https://user-images.githubusercontent.com/43934531/140771079-08ed12ab-1e48-41b2-b729-f54e95ce11a6.png)

We also found out that this problem isn't reproduced in an application with `react-native-windows v0.66.0`. We would like to ask you if it may be a bug, or something has been changed in apps configuration steps and we need to customize some parameters additionally to enable Hermes debugger?

My environment:
Windows 10 Enterprise - 19043.1288
react-native v0.66.0
react-native-windows v0.66.3

**Comment by RedMickey** [comment]
Hi, I found out that this behavior started to reproduce after [this change](https://github.com/microsoft/react-native-windows/commit/1b11bdc61da7680dbe5842060650cb375d8b761e#diff-dc9d488c12414e3d97778c345cd9460fc5fad4dad84ff0aa87ea213a0959f5e8R308-R309). It seems that `m_devSettings->jsiEngineOverride` equals to `JSIEngineOverride::Default` instead of `JSIEngineOverride::Hermes` [here](https://github.com/microsoft/react-native-windows/blob/main/vnext/Shared/OInstance.cpp#L308-L309). The project (RNW v0.66.5) has been configured for Hermes and the `UseHermes` property is set to `true` in the `ExperimentalFeatures.props` file.
I wonder how `JSIEngineOverride` can be set to `JSIEngineOverride::Hermes`. Are some additional configurations required besides initialization with the `npx react-native-windows-init --overwrite --useHermes` to enable Hermes debugger?

**Comment by acoates-ms** [comment]
Can you try connecting using port 9223?

**Comment by RedMickey** [comment]
@acoates-ms I tried to connect to the port 9223, but it seems to be unavailable.
![image](https://user-images.githubusercontent.com/33267199/144076686-cb48ba0f-4a14-457b-8f39-5505051b4f49.png)

Also `localhost:8081` returns an empty array.

I found out that if I change [this condition](https://github.com/microsoft/react-native-windows/blob/react-native-windows_v0.66.7/vnext/Shared/OInstance.cpp#L308-L309) in `react-native-windows` dependency the following way:
```diff
- if (m_devSettings->jsiEngineOverride == JSIEngineOverride::Hermes && m_devSettings->useDirectDebugger && !m_devSettings->useWebDebugger)
+ if (m_devSettings->jsiEngineOverride == JSIEngineOverride::Default && m_devSettings->useDirectDebugger && !m_devSettings->useWebDebugger)
```
then it will be possible to enable Hermes debugger and connect to it from Google Chrome.

**Comment by acoates-ms** [comment]
@mganandraj  - It looks like the logic in [here](https://github.com/microsoft/react-native-windows/blob/main/vnext/Shared/OInstance.cpp#L308-L309) should use some logic to determine the engine being used.  The jsiEngineOverride property shouldn't be used to detect if its hermes, since default might also mean hermes.

**Comment by JeremyBradshaw7** [comment]
I'm also having this issue in 2023 using RN 0.70 !

**Reply by antoniovuono** [reply]
Same here in 0.71.11

**Reply by Hao-yiwen** [reply]
I also have this qs.

**Reply by acoates-ms** [reply]
Can you open an issue with more specific verison details, and repro steps?
